### PR TITLE
executor: migrate test-infra to testify for executor/write_test.go

### LINKED ISSUE #28627: migrate test-infra to testify for `write_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28627

Problem Summary:

### What is changed and how it works?

migrate test-infra to testify for executor/write_test.go 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
PTAL

/cc @tisonkun @time-and-fate @AilinKid

File: executor/write_serial_test.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+	"github.com/pingcap/tidb/util/collate"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdate(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	fillData(tk, "update_test")
```
Comment (tisonkun) [inline executor/write_serial_test.go:35]: If `fillData` has only one usage, I suppose you just inline it.

Comment (hawkingrei) [reply executor/write_serial_test.go:35]: not one usage. There are three test files to use ```fileData```.

File: executor/write_serial_test.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+	"github.com/pingcap/tidb/parser/mysql"
+	"github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/collate"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdate(t *testing.T) {
```
Comment (time-and-fate) [inline executor/write_serial_test.go:30]: Why this case from `testSuite8` becomes a serial test case while others from `testSuite8` are parallel.

Comment (hawkingrei) [reply executor/write_serial_test.go:30]: It is too large for this test case to run it. if parallel, it always fails.

**Review by time-and-fate — APPROVED** [review]
Please resolve the conflicts~

**Comment by hawkingrei** [comment]
> Please resolve the conflicts~

Fixed

File: executor/write_test.go [hunk 3]
```diff
@@ -31,93 +29,91 @@ import (
...
 	tk.MustExec("begin")
 	_, err = tk.Exec(errInsertSelectSQL)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 	tk.MustExec("rollback")
 
 	insertSetSQL := `insert insert_test set c1 = 3;`
 	tk.MustExec(insertSetSQL)
-	tk.CheckLastMessage("")
+	require.Equal(t, tk.Session().LastMessage(), "")
```
Comment (tisonkun) [inline executor/write_test.go:81]: IIRC you may use `require.Empty`

Comment (hawkingrei) [reply executor/write_test.go]: Fixed

**Review by tisonkun — APPROVED** [review]
/merge

**Pull request merged**